tck/src/main/java/org/eclipse/microprofile/concurrency/tck/ThreadContextTest.java

```diff
@@ -580,4 +582,161 @@ public void thirdPartyContextProvidersAreIncludedInThreadContext() {
             Thread.currentThread().setPriority(originalPriority);
         }
     }
+    
+     /**
+     * Verify the MicroProfile Concurrency implementation of propagate(), cleared(), and unchanged()
+     * for ThreadContext.Builder.
+     */
+    @Test
+    public void contextControlsForThreadContextBuilder() throws InterruptedException, ExecutionException, TimeoutException {
+    	ThreadContext allContext = ThreadContext.builder()
+    			.propagated(Buffer.CONTEXT_NAME)
+    			.cleared(Label.CONTEXT_NAME)
+    			.unchanged(THREAD_PRIORITY)
+    			.build();
+    	
+    	try {
+        	ThreadContext.builder()
+			.propagated(Buffer.CONTEXT_NAME)
+			.cleared(Label.CONTEXT_NAME, Buffer.CONTEXT_NAME)
+			.unchanged(THREAD_PRIORITY)
+			.build();
+        	Assert.fail("ThreadContext.Builder.build() should throw an IllegalStateException for set overlap between propagated and cleared");
+    	} catch (IllegalStateException ISE) {
+    		//expected.
+    	}
+
+        int originalPriority = Thread.currentThread().getPriority();
+        try {
+            // Set non-default values
+            int newPriority = originalPriority == 4 ? 3 : 4;
+            Buffer.get().append("contextControls-test-buffer-A");
+            Label.set("contextControls-test-label-A");
+
+            //Future<Integer> future1 = unmanagedThreads.submit(
+            		
+            Callable<Integer> callable = allContext.contextualCallable(() -> {
+            	Assert.assertEquals(Buffer.get().toString(), "contextControls-test-buffer-A-B",
+            			"Context type was not propagated to contextual action.");
+
+            	Buffer.get().append("-C");
+
+            	Assert.assertEquals(Label.get(), "",
+            			"Context type that is configured to be cleared was not cleared.");
+
+            	Label.set("contextControls-test-label-C");
+            	
+            	return Thread.currentThread().getPriority();
+            });
+            
+            Buffer.get().append("-B");
+            Label.set("contextControls-test-label-B");
+
+            Future<Integer> future = unmanagedThreads.submit(() -> {
+            		Buffer.get().append("unpropagated-buffer");
+            		Label.set("unpropagated-label");
+            		Thread.currentThread().setPriority(newPriority);
+            		return callable.call();
+            });
+
+            Assert.assertEquals(future.get(MAX_WAIT_NS, TimeUnit.NANOSECONDS), Integer.valueOf(newPriority),
+                    "Callable returned incorrect value.");
+            
+            Assert.assertEquals(Buffer.get().toString(), "contextControls-test-buffer-A-B-C",
+            		"Context type was not propagated to contextual action.");
+
+            Assert.assertEquals(Label.get(), "contextControls-test-label-B",
+            		"Context type was not propagated to contextual action.");
```

Comment:
I think you forgot to update this failure message after copy/paste from elsewhere.  I'm not sure what to put for the message here though, because as far as I can tell it doesn't really test anything. This label was set onto the current thread earlier, but the current thread didn't do anything of consequence after that point which would cause it to change (all of the propagation work is done on a separate, unmanaged thread).